### PR TITLE
[DEV-6490] Assurance Statement url

### PR DESCRIPTION
**Pull request by collinwr**
**Description:**
This PR adds an `assurance_statement_url` reporting agency overview endpoints.

Example url: https://files.usaspending.gov/agency_submissions/Raw%20DATA%20Act%20Files/2020/P10/073%20-%20Small%20Business%20Administration%20(SBA)/2020-P10-073_Small%20Business%20Administration%20(SBA)-Assurance_Statement.txt

**Technical details:**
URL is built using the following fields:  
- agency_name
- abbreviation
- toptier_code
- fiscal_year
- fiscal_period
- submission_is_quarter

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
    - [x] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. N/A Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-6490](https://federal-spending-transparency.atlassian.net/browse/DEV-6490):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/agency/v2/views/agency_base.py [hunk 1]
```diff
@@ -78,6 +83,36 @@ def validate_fiscal_period(request_data):
...
+        agency_name, abbreviation, toptier_code, fiscal_year, fiscal_period, submission_is_quarter
+        """
+        try:
+            agency_name_split = result["agency_name"].split(" ")
+            abbreviation_wrapped = f"({result['abbreviation']})"
+            toptier_code = result["toptier_code"]
+            fiscal_year = result["fiscal_year"]
+
+            if result["submission_is_quarter"]:
+                fiscal_period = f"Q{int(result['fiscal_period'] / 3)}"
```
Comment (emily-brents) [inline usaspending_api/agency/v2/views/agency_base.py:99]: We have a `get_quarter_from_period()` method in fiscal_year_helpers, which could be useful here.

File: usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/overview.md [hunk 2]
```diff
@@ -133,3 +135,4 @@ This endpoint returns an overview of government agency submission data.
     The difference in file A and file B obligations.
 + `unlinked_contract_award_count` (required, number)
 + `unlinked_assistance_award_count` (required, number)
++ `assurance_statement_url` (required, number)
```
Comment (emily-brents) [inline usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/overview.md:138]: Change from number to string

File: usaspending_api/api_contracts/contracts/v2/reporting/agencies/overview.md [hunk 3]
```diff
@@ -135,3 +137,4 @@ This endpoint returns an overview list of government agencies submission data.
     The difference in File A and File B obligations.
 + `unlinked_contract_award_count` (required, number)
 + `unlinked_assistance_award_count` (required, number)
++ `assurance_statement_url` (required, number)
```
Comment (emily-brents) [inline usaspending_api/api_contracts/contracts/v2/reporting/agencies/overview.md:140]: `number` again

**Review by ebdabbs — COMMENTED** [review]
Once @emily-brents comments are resolved, API contract changes are good!

File: usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/overview.md (hunk 2 above)
Comment (collinwr) [reply usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/overview.md:138]: Updated

File: usaspending_api/api_contracts/contracts/v2/reporting/agencies/overview.md (hunk 3 above)
Comment (collinwr) [reply usaspending_api/api_contracts/contracts/v2/reporting/agencies/overview.md:140]: Updated

File: usaspending_api/agency/v2/views/agency_base.py (hunk 1 above)
Comment (collinwr) [reply usaspending_api/agency/v2/views/agency_base.py:99]: The method you listed did not meet the needs here exactly, so I left it as is

Comment (emily-brents) [reply usaspending_api/agency/v2/views/agency_base.py:99]: fair enough

**Review by emily-brents — DISMISSED** [review]
Tested locally, LGTM

**Review by ebdabbs — APPROVED** [review]
API contract changes approved

**Review by emily-brents — APPROVED** [review]
LGTM

**Pull request merged**